packages/cli/tests/e2e/multienv/TestRemoteHappyPath.tests.ts

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import fs from "fs-extra";
+import path from "path";
+
+import {
+  AadValidator,
+  BotValidator,
+  FrontendValidator,
+  FunctionValidator,
+  SimpleAuthValidator,
+} from "../../commonlib";
+
+import {
+  cleanUp,
+  execAsync,
+  execAsyncWithRetry,
+  getSubscriptionId,
+  getTestFolder,
+  getUniqueAppName,
+  mockTeamsfxMultiEnvFeatureFlag,
+  setSimpleAuthSkuNameToB1Bicep,
+} from "../commonUtils";
+import AppStudioLogin from "../../../src/commonlib/appStudioLogin";
+import { deserializeDict } from "@microsoft/teamsfx-core";
+import {
+  err,
+  FxError,
+  Result,
+  ok,
+  ConfigFolderName,
+  PublishProfilesFolderName,
+  EnvProfileFileNameTemplate,
+  EnvNamePlaceholder,
+} from "@microsoft/teamsfx-api";
+import { expect } from "chai";
+
+// Load envProfile with userdata (not decrypted)
+async function loadContext(projectPath: string, env: string): Promise<Result<any, FxError>> {
+  const context = await fs.readJson(
+    path.join(
+      projectPath,
+      `.${ConfigFolderName}`,
+      PublishProfilesFolderName,
+      EnvProfileFileNameTemplate.replace(EnvNamePlaceholder, env)
+    )
+  );
+  const userdataContent = await fs.readFile(
+    path.join(projectPath, `.${ConfigFolderName}`, PublishProfilesFolderName, `${env}.userdata`),
+    "utf8"
+  );
+  const userdata = deserializeDict(userdataContent);
+
+  const regex = /\{\{([^{}]+)\}\}/;
+  for (const component in context) {
+    for (const key in context[component]) {
+      const matchResult = regex.exec(context[component][key]);
+      if (matchResult) {
+        const userdataKey = matchResult[1];
+        if (userdataKey in userdata) {
+          context[component][key] = userdata[key];
+        }
+      }
+    }
+  }
+  return ok(context);
+}
+
+describe("Create single tab/bot/function", function () {
+  const env = "e2e";
+  const testFolder = getTestFolder();
+  const appName = getUniqueAppName();
+  const subscription = getSubscriptionId();
+  const projectPath = path.resolve(testFolder, appName);
+  const processEnv = mockTeamsfxMultiEnvFeatureFlag();
+
+  it(`Happy path`, async function () {
+    // new a project (tab + bot + function)
+    try {
+      let result;
+      result = await execAsync(
+        `teamsfx new --interactive false --app-name ${appName} --capabilities tab bot --azure-resources function --programming-language javascript`,
+        {
+          cwd: testFolder,
+          env: processEnv,
+          timeout: 0,
+        }
+      );
+      console.log(
+        `[Successfully] scaffold to ${projectPath}, stdout: '${result.stdout}', stderr: '${result.stderr}''`
```

Comment:
I think stdout may be too much and hope not to print it if successfully.

Reply:
I think too much log is better than too little log. Especially for E2E, running a single test takes too much time. The effort of retrying the test and print the debug info is big. And I don't see why too much log will cause an issue here.